sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/RntbdTransportClient.java

```diff
@@ -373,18 +524,18 @@ private Options(final Builder builder) {
         private Options(final ConnectionPolicy connectionPolicy) {
             this.bufferPageSize = 8192;
             this.connectionAcquisitionTimeout = Duration.ofSeconds(5L);
+            this.connectionEndpointRediscovery = connectionPolicy.isTcpConnectionEndpointRediscoveryEnabled();
             this.connectTimeout = connectionPolicy.getConnectTimeout();
             this.idleChannelTimeout = connectionPolicy.getIdleTcpConnectionTimeout();
             this.idleChannelTimerResolution = Duration.ofMillis(100);
-            this.idleEndpointTimeout = connectionPolicy.getIdleEndpointTimeout();
+            this.idleEndpointTimeout = connectionPolicy.getIdleTcpEndpointTimeout();
```

Comment:
wonder should we make the name more consistent. Like using idleTcpEndpointTimeout across board.